### PR TITLE
[INV-107] Create endpoint to get categorized bookings

### PR DESCRIPTION
**Pull request by Savesh1508**
# Pull request for: [INV-107]

## Description

- Added migration in which booking statuses are improved
- Added endpoint for getting user bookings
- Added util to categorize user bookings

Screenshot_2

## Checklist

- [x] I have manually tested the changes and related features
- [x] I have run the linter and fixed any possible issues
- [x] I have run the build and fixed any possible issues

### CODE REVIEW

File: database/migrations/1733797260473-fixed-booking-status.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class FixedBookingStatus1733797260473 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      ALTER TYPE "booking_status_enum"
+      ADD VALUE 'cancelled';
```
Comment (ShahobiddinIsmoilov) [inline database/migrations/1733797260473-fixed-booking-status.ts:7]: Please use the american version ```canceled```. In other parts of the project we are using it. Let's keep consistency.

File: src/modules/bookings/bookings.controller.ts [hunk 2]
```diff
@@ -23,4 +24,23 @@ export class BookingsController {
...
+  @Get()
+  @ApiBearerAuth()
+  @UseGuards(AccessTokenGuard)
+  @ApiOperation({ summary: 'Get user bookings' })
+  @ApiResponse({
+    status: 200,
+    description: 'User bookings successfully received.',
+  })
+  @ApiResponse({ status: 401, description: 'Unauthorized request.' })
+  async getUserBookings(@GetAccount('accountId') userId: string) {
```
Comment (ShahobiddinIsmoilov) [inline src/modules/bookings/bookings.controller.ts:37]: I think ```accountId``` is more logical. Also, we are using ```accountId``` in other places

File: src/modules/bookings/utils/bookings-categorize.util.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export function categorizeBookings(bookings: Booking[]) {
+  const categorized: CategorizedBookings = {
+    pending: [],
+    active: [],
+    inactive: [],
+    upcoming: [],
+    completed: [],
+    cancelled: [],
```
Comment (ShahobiddinIsmoilov) [inline src/modules/bookings/utils/bookings-categorize.util.ts:15]: ```canceled```

File: src/shared/constants/booking-status.constant.ts [hunk 4]
```diff
@@ -4,4 +4,5 @@ export enum BookingStatus {
   Pending = 'pending',
   Completed = 'completed',
   Upcoming = 'upcoming',
+  Cancelled = 'cancelled',
```
Comment (ShahobiddinIsmoilov) [inline src/shared/constants/booking-status.constant.ts:7]: The same here

**Review by ShahobiddinIsmoilov — CHANGES_REQUESTED** [review]

File: database/migrations/1733797260473-fixed-booking-status.ts (hunk 1 above)
Comment (Savesh1508) [reply database/migrations/1733797260473-fixed-booking-status.ts:7]: Fixed

File: src/modules/bookings/bookings.controller.ts (hunk 2 above)
Comment (Savesh1508) [reply src/modules/bookings/bookings.controller.ts:37]: Fixed

File: src/modules/bookings/utils/bookings-categorize.util.ts (hunk 3 above)
Comment (Savesh1508) [reply src/modules/bookings/utils/bookings-categorize.util.ts:15]: Fixed

File: src/shared/constants/booking-status.constant.ts (hunk 4 above)
Comment (Savesh1508) [reply src/shared/constants/booking-status.constant.ts:7]: Fixed

**Review by ShahobiddinIsmoilov — APPROVED** [review]

**Review by farrooh — APPROVED** [review]

**Pull request merged**